includes/class-sensei-guest-user.php

```diff
@@ -466,7 +476,28 @@ private function is_action( $field, $nonce ) {
 	 */
 	public function skip_sensei_email( $send_email ) {
 		return $this->is_current_user_guest() ? false : $send_email;
+	}
 
+	/**
+	 * Prevent emails related to the guest user from being dispatched via wp_mail.
+	 *
+	 * @access private
+	 * @since  $$next-version$$
+	 *
+	 * @param bool|null $return Null if we should send the email, a boolean if not.
+	 * @param array     $atts   Email attributes.
+	 * @return bool|null Null if we should send the email, a boolean if not.
+	 */
+	public function skip_wp_mail( $return, $atts ) {
+		if ( $this->is_current_user_guest() ) {
+			// If this e-mail is being dispatched while the current user is a guest, just... don't send it.
+			return false;
+		}
+		if ( Sensei_Temporary_User::should_block_email( $atts, self::EMAIL_DOMAIN ) ) {
+			// If this e-mail is being dispatched to a guest user, don't send it.
+			return false;
+		}
```

Comment:
In this block, we're returning `false` so `wp_mail` returns `false` too. This is considered an "error". It might not be the most friendly way of doing this (for any third party developer debugging this), but I believe it is better than returning `true` and faking a possible success..